auth/handler.go

```diff
@@ -305,8 +318,9 @@ func (h *Handler) handleProvider(id string, p IdentityProvider, refU *url.URL, w
 		return
 	}
 
-	var userID string
-	err = h.userLookup.QueryRowContext(ctx, id, sub.SubjectID).Scan(&userID)
+	if sub.UserID == "" {
+		err = h.userLookup.QueryRowContext(ctx, id, sub.SubjectID).Scan(&sub.UserID)
+	}
```

Comment:
I'm curious why we would want to look up a user if the user is set to empty. Could we skip the lookup entirely since we know the userID isn't present and return an appropriate error?

Reply:
`sub` is returned from the IdentityProvider. Most providers only know of the subject ID, (e.g. the GitHub ID for GitHub auth) and so we need to find which GoAlert UserID that maps to -- unless we already know the user, as with the `authlink` provider.